src/test/record/records_service/test_response_time_all.py

```diff
@@ -139,6 +139,383 @@ def test_drop_case(self):
+        records = create_records(records_raw, columns)
+
+        response_time = ResponseTime(records)
+
+        expect_raw = [
+            {'start': 0, 'response_time': 2},
+            {'start': 3, 'response_time': 1},
+            {'start': 11, 'response_time': 1}
+        ]
+        result = to_dict(response_time.to_best_case_records())
+        assert result == expect_raw
+
+    def test_three_column_default_case(self):
+        records_raw = [
+            {'start': 0, 'middle': 1, 'end': 2},
+            {'start': 3, 'middle': 4, 'end': 6},
+            {'start': 11, 'middle': 13, 'end': 16}
+        ]
+        columns = [ColumnValue('start'), ColumnValue('middle'), ColumnValue('end')]
+        records = create_records(records_raw, columns)
+
+        response_time = ResponseTime(records)
+
+        expect_raw = [
+            {'start': 0, 'response_time': 2},
+            {'start': 3, 'response_time': 3},
+            {'start': 11, 'response_time': 5}
+        ]
+        result = to_dict(response_time.to_best_case_records())
+        assert result == expect_raw
+
+    def test_single_input_multi_output_case(self):
+        records_raw = [
+            {'start': 0, 'middle': 4, 'end': 5},
+            {'start': 0, 'middle': 4, 'end': 6},
+            {'start': 0, 'middle': 12, 'end': 13}
+        ]
+        columns = [ColumnValue('start'), ColumnValue('middle'), ColumnValue('end')]
+        records = create_records(records_raw, columns)
+
+        response_time = ResponseTime(records)
+
+        expect_raw = [
+            {'start': 0, 'response_time': 5}
+        ]
+        result = to_dict(response_time.to_best_case_records())
+        assert result == expect_raw
+
+    def test_multi_input_single_output_case(self):
+        records_raw = [
+            {'start': 0, 'middle': 4, 'end': 13},
+            {'start': 1, 'middle': 4, 'end': 13},
+            {'start': 5, 'middle': 12, 'end': 13}
+        ]
+        columns = [ColumnValue('start'), ColumnValue('middle'), ColumnValue('end')]
+        records = create_records(records_raw, columns)
+
+        response_time = ResponseTime(records)
+
+        expect_raw = [
+            {'start': 5, 'response_time': 8}
+        ]
+        result = to_dict(response_time.to_best_case_records())
+        assert result == expect_raw
+
+    def test_cross_flow_case(self):
+        records_raw = [
+            {'start': 0, 'end': 10},
+            {'start': 3, 'end': 4},
+            {'start': 4, 'end': 10},
+            {'start': 6, 'end': 6},
+        ]
+        columns = [ColumnValue('start'), ColumnValue('end')]
+
+        records = create_records(records_raw, columns)
+        response = ResponseTime(records)
+
+        response_time = response.to_best_case_records()
+        expect = [
+            {'start': 3, 'response_time': 1},
+            {'start': 4, 'response_time': 6},
+            {'start': 6, 'response_time': 0}
+        ]
+        assert to_dict(response_time) == expect
```

Comment:
@rokamu623 
Is it poasible that the start_ts is bigger than end_ts?
If the case is possible, I think you should add the test.